openwisp_monitoring/device/api/views.py

```diff
@@ -165,6 +167,12 @@ def post(self, request, pk):
         except ValidationError as e:
             logger.info(e.message)
             return Response(e.message, status=status.HTTP_400_BAD_REQUEST)
+        time_obj = request.query_params.get('time')
+        if time_obj:
+            time = datetime.strptime(time_obj, '%d-%m-%Y_%H:%M:%S.%f')
```

Comment:
what if a bogus time is supplied?
Please make sure 400 bad request is returned with a proper error message + a test.